### PR TITLE
Update to go-ipfs@0.4.19

### PR DESCRIPTION
**Pull request by sanderpick**
Bunch of changes required... we should take this for a spin with Photos Beta for a bit.

### CODE REVIEW

**Comment by sanderpick** [comment]
Gonna merge this, not really reviewable... quick approval @carsonfarmer ?

File: ipfs/util.go [hunk 1]
```diff
@@ -100,6 +78,28 @@ func ShortenID(id string) string {
 	return id[len(id)-7:]
 }
 
+// IpnsSubs shows current name subscriptions
+func IpnsSubs(node *core.IpfsNode) ([]string, error) {
```
Comment (carsonfarmer) [inline ipfs/util.go]: ooh, this is nice

File: keypair/full.go [hunk 2]
```diff
@@ -69,10 +69,10 @@ func (kp *Full) LibP2PPubKey() (*libp2pc.Ed25519PublicKey, error) {
 }
 
 func (kp *Full) Verify(input []byte, sig []byte) error {
-	if len(sig) != 64 {
+	if len(sig) != ed25519.PrivateKeySize {
 		return ErrInvalidSignature
 	}
-	var asig [64]byte
+	var asig [ed25519.PrivateKeySize]byte
```
Comment (carsonfarmer) [inline keypair/full.go]: sweet

File: ipfs/crypto.go [hunk 3]
```diff
@@ -34,6 +37,32 @@ func IdentityConfig(sk libp2pc.PrivKey) (config.Identity, error) {
 	return ident, nil
 }
 
+// UnmarshalPrivateKey converts a protobuf serialized private key into its
+// representative object
+func UnmarshalPrivateKey(data []byte) (libp2pc.PrivKey, error) {
```
Comment (sanderpick) [inline ipfs/crypto.go]: For future, may as well _always_ use this to unmarshal our ed25519 keys for backwards compat.

Comment (carsonfarmer) [reply ipfs/crypto.go]: Yeh good call. I can also port this over the JS land when required.

**Pull request merged**